tests/test.spec.ts

```diff
@@ -1,18 +1,78 @@
 import { join } from 'path';
 import * as rimraf from 'rimraf';
 import * as shell from 'shelljs';
+import { newImportStatementRegex, newStringRegex } from '../dist/utils';
 
 const projectsRoot = join(__dirname, '../projects');
 
-[1, 3, 4, 5, 6, 7, 8].forEach((value) => {
-  it(`Project ${value}`, () => {
-    const projectDir = join(projectsRoot, `project${value}`);
-    rimraf.sync(join(projectDir, 'dist'));
-    const { code, stderr } = shell.exec('npm start', {
-      cwd: projectDir,
-      silent: true
-    });
-    if (code !== 0) console.error(stderr);
-    expect(code).toEqual(0);
+/**
+ * Sample JavaScript code with a bunch of require/import
+ * statements with various spacing etc. Import paths from
+ * valid statements are incrementing numbers, starting from 0,
+ * so that it is easy to verify validity of results.
+ */
+const sampleImportStatements = `
+const module = require('0')
+var module = require
+(
+  '1'
+)  ;
+import module from ('2');
+import "3"
+
+imported ("invalid/import")
+
+import theDefault, {namedExport} from ("4")
+import {
+  extraLinesOhNo
+} from '5'
+const asyncImport = await import('6');
+
+export * from '7';
+
+import
+
+  '8'
+
+const notAnImport = unimport('something');
+`;
+
+function runTestProject(projectNumber: number) {
+  const projectDir = join(projectsRoot, `project${projectNumber}`);
+  rimraf.sync(join(projectDir, 'dist'));
+  const { code, stderr } = shell.exec('npm start', {
+    cwd: projectDir,
+    silent: true
+  });
+  if (code !== 0) console.error(stderr);
+  expect(code).toEqual(0);
+}
+
+it(`Import regex matches import statements`, () => {
+  const expectedImportPaths = sampleImportStatements.match(
+    /(\d+)/g
+  ) as string[];
+
+  const importStatementMatches = sampleImportStatements.match(
+    newImportStatementRegex('g')
+  );
+  expect(importStatementMatches).toHaveLength(expectedImportPaths.length);
+
+  const foundImportPaths: string[] = [];
+  for (const importStatement of importStatementMatches) {
+    // Global match is a string, not a match group, so re-match without the global flag.
+    const pathMatch = importStatement.match(newStringRegex());
+    expect(pathMatch).toBeTruthy();
+    foundImportPaths.push(pathMatch.groups.path);
+  }
+  expectedImportPaths.forEach((expectedPath, i) => {
+    expect(expectedPath).toEqual(foundImportPaths[i]);
+  });
+});
+
+// Run tests on projects. 9-11 are for testing fullpath file resolution
```

Comment:
Manual review of the resulting files for the new 9-11 projects (which use the new flag) gave the expected results. I'm not sure what the coverage of import statement cases is with these tests, but between the direct testing of the regex used and these tests to ensure that paths can be fully resolved, it should be sufficient coverage.